### ISSUE TITLE
Asymmetric routing issues with additional CIDR blocks 

### ISSUE THREAD

**Issue by causton81** [aws/amazon-vpc-cni-k8s#1605]
**What would you like to be added**:
**TL;DR**: the Debian AMI that I used for my node runs dhclient on all interfaces on bootup. Sometimes the default route and hostname get messed up by the dhclient of the extra ENI. I changed my network config to only run dhcp on the primary ENI (I copied the cloud-config generated ens5 config to `/etc/network/interfaces.d/ens5` and then commented out `/etc/network/interfaces:#source-directory /run/network/interfaces.d`. This seems to be working a lot better and seems to be what the aws-vpc-cni driver expects from what I can grok in the iptables mangle+nat tables and `ip rule`s.

I have a VPC with a small primary CIDR within 10..0/8 and a secondary CIDR under 100.81.128.0/18 per rules at https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#add-cidr-block-restrictions

My company runs a 10..0/8 private corporate network, so I was trying to leave open the option of adding a CIDR block from our networking team in the future (for use with a transit gateway and VPN back to headquarters).

I put a jump host on a 10/8 subnet and a Kubernetes api/worker node on a 100.81.128.0/18 subnet. I used kubeadm to stand up kubernetes and helm to install eks/aws-vpc-cni.

It seemed to work ok until I came into work the next day and could not ssh into the kubernetes node from my jump host. I finally tracked this down to my response packets leaving on the extra ENI interface added by the CNI driver instead of the interface the request came from. Security Groups are stateful and the groups involved in this connection could not tie the request and response back together because of the different interfaces.

My best guess is that the dhcp clients race to set the default gateway in the main route table.

I'm curious if this is documented anywhere or could be handled by the aws-vpc-cni somehow? I can take a crack at adding something to the README if the maintainers think it's worth doing.

~~I worked around it (I think) with policy based routing and a quick bash script:~~
```bash
# removed by author; script did not solve all issues
```

```
root@ip-100-81-130-215:~# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@ip-100-81-130-215:~# uname -a
Linux ip-100-81-130-215 5.10.0-0.bpo.4-amd64 #1 SMP Debian 5.10.19-1~bpo10+1 (2021-03-13) x86_64 GNU/Linux
root@ip-100-81-130-215:~#
```

**Why is this needed**:

This AWS [documentation](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html#add-cidr-block-restrictions) describes combinations of CIDR blocks on a VPC. People experimenting with multiple CIDR blocks and this CNI driver are highly likely to encounter asymmetric routing and it is somewhat difficult to diagnose (at least it was for me).

**Comment by causton81** [comment]
I think I found the problem of multiple dhclients and added a TL;DR to original description. Maybe it would still be worth adding a warning to the readme?

**Comment by jayanthvn** [comment]
This looks similar to this issue - https://github.com/coreos/bugs/issues/992?
